src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3531,6 +3531,11 @@ public override SyntaxNode ExpressionStatement(SyntaxNode expression)
             return SyntaxFactory.ExpressionStatement((ExpressionSyntax)expression);
         }
 
+        public override SyntaxNode ConditionalAccessExpression(SyntaxNode expression, SyntaxNode whenNotNull)
+        {
+            return SyntaxFactory.ConditionalAccessExpression(Parenthesize(expression), SyntaxFactory.MemberBindingExpression((SimpleNameSyntax)whenNotNull));
```

Comment:
Yes.  Your code generation logic is wrong :)   As kevin mentioned, you should not be parenthesizing the whenNotNull portion.